frontend/src/pages/FindChain.tsx

```diff
@@ -144,14 +143,31 @@ const FindChain = () => {
 
   //get search term from input
   const onChange = (e: any) => {
-    setSearchTerm(e.target.value);
+    let searchTerm: string = e.target.value;
 
+    //if no search term, show nothing
+    if (searchTerm == ""){
+      return setValue(null);
+    }
+
+    //find matches
+    let matches: IChain[] = [];
+    let matchesLength: number = 0;
     chainData.filter((val) => {
-      if (searchTerm == "") {
-      } else if (val.name.toLowerCase().includes(searchTerm.toLowerCase())) {
-        return setValue(val);
+      if (val.name.toLowerCase().includes(searchTerm.toLowerCase())) {
+        matches.push(val);
+        matchesLength++;
```

Comment:
Please use .filter correctly: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

matches = chainData.filter(val => val.name.toLowerCase().includes(searchTerm.toLowerCase()));

Reply:
Thank you for the correction. I go with the find method here then.